science/pdb2pqr/Portfile

```diff
@@ -1,67 +1,50 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
-PortSystem          1.0
-PortGroup           python 1.0
+PortSystem              1.0
+PortGroup               python 1.0
+PortGroup               github 1.0
 
-name                pdb2pqr
-version             2.1.1
-revision            1
-categories          science
-maintainers         {gmail.com:howarth.at.macports @jwhowarth}
-description         automate Poisson-Boltzmann electrostatics calculations
-long_description    PDB2PQR is a Python software package that automates many of the \
-                    common tasks of preparing structures for continuum electrostatics \
-                    calculations, providing a platform-independent utility for converting \
-                    protein files in PDB format to PQR format.
-homepage            http://pdb2pqr.sourceforge.net/
-platforms           darwin
-master_sites        sourceforge:project/pdb2pqr/pdb2pqr/pdb2pqr-${version}/
-distfiles           ${name}-src-${version}.tar.gz
-checksums           rmd160  b08fc9118cc3b96f57af510829e58af14bbefe16 \
-                    sha256  c2b14b0c95ffd76c910af108541a34beaea5f5c2e44246a61640644b2990c3cb \
-                    size    14711417
+name                    pdb2pqr
+revision                0
 
-python.default_version  27
+supported_archs         noarch
+platforms               {darwin any}
+license                 BSD
+maintainers             {gmail.com:howarth.at.macports @jwhowarth} openmaintainer
 
-worksrcdir          ${name}-src-${version}
+description             Preparation of protein structures for electrostatics calculations
+long_description        {*}${description}
 
-depends_lib         port:py${python.version}-numeric
+# The PyPi copy of pdb2pqr has its testsuite in DOS format which
+# makes it unpatchable under unix. So use the pdb2pqr from the 
+# Electrostatics GitHub
 
-post-patch {
-    fs-traverse f ${worksrcpath} {
-        if {[file isfile ${f}] && [file extension ${f}] eq ".py"} {
-            reinplace -E "s,(/usr/bin/python|/usr/bin/env python),${prefix}/bin/python${python.branch},g" ${f}
-        }
-    }
-}
+github.setup            Electrostatics pdb2pqr 3.6.1 v
+github.tarball_from     releases
+distfiles               pdb2pqr-v${version}.tar.gz:pdb2pqr
 
-build.cmd           ${prefix}/bin/python${python.branch} scons/scons.py
-build.target
-build.args          BUILD_PDB2PKA=True
+checksums               rmd160  bdd66b7cc6ecf069c709bb5b8eb0665ec42602d2 \
+                        sha256  48ab09877e9c35dad4c46459ad31422013ea153e3681fc2cace5bf68ce352d75 \
+                        size    13389894
 
-test.run            yes
-test.cmd            ${prefix}/bin/python${python.branch} scons/scons.py
-test.target         complete-test
+python.versions         312
```

Comment:
as for the other port, this doesn't do anything for port that are not named `py-<portname>` - please use `python.default_version 312` instead.